ios/dtx_codec/channel.go

```diff
@@ -136,6 +155,12 @@ func (d *Channel) Dispatch(msg Message) {
 			return
 		}
 	}
+	if msg.PayloadHeader.MessageType == UnknownTypeOne {
+		d.mutex.Unlock()
+		d.messageReceiver.msgChannel <- msg
+
+		return
+	}
```

Comment:
`Dispatcher` implementations are getting those messages already atm, and we would change that with this filter here. Lets continue to send them to the `Dispatcher`, in our case now the `GlobalDispatcher`.

Since `GlobalDispatcher` only handles very few messages, we would need something that forwards it from there, or allow to pass in a custom `Dispatcher` for the global channel

Reply:
according to the logs, it doesn't look like that it handles few messages, it is enormous amount of them, I tried already putting the logic in `Dispatcher` only :D 

it results in the problem here:
```
1 136 129 1 137 129 1 138 129 1 139 129 1 140 129 1 141 129 1 142 129 1 143 129 1 144 129 1 145 129 1 146 129 1 147 129 1 148 129 1 149 129 1 150 129 1 151 129 1 152 129 1 153 129 1 154 129 1 155 129 1 156 129 1 157 129 1 158 129 1 159 129 1 160 129 1 161 129 1 162 129 1 163 129 1 164 129 1 165 129 1 166 129 1 167 129 1 168 129 1 169 129 1 170 129 1 171 129 1 172 129 1 173 129 1 174 129 1 175 129 1 176 129 1 177 129 1 178 129 1 179 129 1 180 129 1 181 129 1  216 3 217 3 218 3 219 3 220 3 221 3 222 3 223 3 224 3 225 3 226 3 227 3 228 3 229 3 230 3 231 3 232 3 233 3 234 3 235 3 236 3 237 3 238 3 239 3 240 3 241 3 242 3 243 3 244 3 245 3 246 3 247 3 248 3 249 3 250 3 251 3 252 3 253 3 254 3 255 4 0 4 1 4 2 4  0 0 0 0 0 0 4 2 0 0 0 0 0 0 126 189 0 0 0 0 0 0 0 0 0 0 0 0 0 2 200 208]","level":"fatal","msg":"Start call error","time":"2024-11-01T10:09:30-05:00"}

```

you generally receive infinite bytes stream and then ends up with the error. That is why I provided the solution as it is now in the MR :) 

I checked, with this current code, it has no negative effect on our side executions as this sysmontap gets only desired message and then we close the connection and the channel

Reply:
> I checked, with this current code, it has no negative effect on our side executions as this sysmontap gets only desired message and then we close the connection and the channel

Since `UnknownTypeOne` was added during the work of XCUITest for iOS 17 I strongly assume that a message with that type is sometimes sent during a test execution. And if that happens, we would block here forever as `d.messageReceiver.msgChannel` is `nil`. Pretty much any occasion of a message of that type would block forever, unless something calls `Channel.Receive`, but that would also only help on the global channel.

That problem is solved with having it in the `Dispatcher` interface as unhandled messages are only thrown away there.

We would need something in this method https://github.com/danielpaulus/go-ios/blob/6fa8b80d4a03721bc37ac1fc466465145cb5adec/ios/dtx_codec/connection.go#L91 that forwards the messages that were not handled in this method to another `Dispatcher` (or allow to set a customer `Dispatcher` on the global channel)

Reply:
alright, the latest commit should safely avoid potential issues with XCTests